lambdas/authorizer/src/index.ts

```diff
@@ -1,21 +1,95 @@
+// eslint-disable no-console
+
 import type { APIGatewayTokenAuthorizerHandler } from 'aws-lambda';
+import { z } from 'zod';
+import {
+  CognitoIdentityProviderClient,
+  GetUserCommand,
+} from '@aws-sdk/client-cognito-identity-provider';
+import { jwtDecode } from 'jwt-decode';
+import { logger } from './logger';
+
+const cognitoClient = new CognitoIdentityProviderClient({
+  region: 'eu-west-2',
+});
+
+const $AccessToken = z.object({
+  client_id: z.string(),
+  iss: z.string(),
+  token_use: z.string(),
+});
+
+const generatePolicy = (Resource: string, Effect: 'Allow' | 'Deny') => ({
+  principalId: 'api-caller',
+  policyDocument: {
+    Version: '2012-10-17',
+    Statement: [
+      {
+        Action: 'execute-api:Invoke',
+        Effect,
+        Resource,
+      },
+    ],
+  },
+});
+
+const getEnvironmentVariable = (envName: string) => process.env[envName];
+
+export const handler: APIGatewayTokenAuthorizerHandler = async ({
+  methodArn,
+  authorizationToken,
+}) => {
+  try {
+    const userPoolId = getEnvironmentVariable('USER_POOL_ID');
+    const userPoolClientId = getEnvironmentVariable('USER_POOL_CLIENT_ID');
+
+    if (!userPoolId || !userPoolClientId) {
+      logger.error('Lambda misconfiguration');
+      return generatePolicy(methodArn, 'Deny');
+    }
+
+    const decodedToken = jwtDecode(authorizationToken);
+
+    const {
+      client_id: clientId,
+      iss,
+      token_use: tokenUse,
+    } = $AccessToken.parse(decodedToken);
```

Comment:
The token should have an expiry time and should be rejected if it is expired (although maybe that would happen before it gets here?)

Reply:
we are now verifying the token using `jwks-rsa`, which includes an expiry check